src/App.tsx

```diff
@@ -1,51 +1,80 @@
 import React, { useState, useEffect } from 'react';
+import { HashRouter as Router, Routes, Route } from 'react-router-dom';
 import { ThemeProvider, CssBaseline, Box } from '@mui/material';
 import { lightTheme, darkTheme } from './themes';
 import Sidebar from './components/Sidebar';
-import Main from './components/Main';
 import Bottom from './components/Bottom';
+import ErrorSnackbar from './components/ErrorSnackbar';
+import Main from './components/Main';
+import Cosimulation from './components/Cosimulation/Cosimulation';
 
 const App: React.FC = () => {
   const [darkMode, setDarkMode] = useState(false);
+  const [sidebarOpen, setSidebarOpen] = useState(true);
+  const sidebarWidth = sidebarOpen ? 240 : 64;
+
+  const toggleDarkMode = () => setDarkMode((prev) => !prev);
+  const toggleSidebar = () => setSidebarOpen((prev) => !prev);
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth < 768) {
```

Comment:
At the current state, the Sidebar is collapsed when the width is lower than the value, but it can reopened if needed. Is it supposed to disappear completely?

Reply:
Collapsable option is good. Please keep it.